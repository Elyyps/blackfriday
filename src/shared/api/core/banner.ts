import { IBannerComponentProps } from "@app/core/banner";
import CoolBlue from "@assets/cool-blue.svg";
import Oasis from "@assets/oasis.jpg";

export const bannerProps: IBannerComponentProps = {
  title: "Black Friday bij Coolblue",
  buttonTitle: "Meer deals",
  buttonLink: "#",
  image: { src: Oasis, alt: "alt text oasis" },
  label: "22 t/m 26 nov.",
  text: "Een selectie van bizarre aanbiedingen. Meer info",
  logo: { src: CoolBlue, alt: "alt text CoolBlue" }
};
