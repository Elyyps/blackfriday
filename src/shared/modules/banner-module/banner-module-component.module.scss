.banner-module {
  display: initial;
}

.container {
  width: 100%;
  @media (max-width: $breakpoint-small) {
    width: auto;
  }
}

.holder {
  background: $global-background;
  border-radius: $global-border-radius;
  color: $gray-light;
  overflow: hidden;

  @media (min-width: $breakpoint-small) {
    display: flex;
  }
}
