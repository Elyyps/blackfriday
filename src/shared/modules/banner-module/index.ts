export * from "./banner-module.component";
