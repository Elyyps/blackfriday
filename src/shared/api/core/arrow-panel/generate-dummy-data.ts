import { IArrowPanelProps } from "@app/core/arrow-panel";
import party from "@assets/icons/party.svg";
import iears from "@assets/dummy-images/product-thumbnails/iears.png";
import imac from "@assets/dummy-images/product-thumbnails/imac.png";
import iphone from "@assets/dummy-images/product-thumbnails/iphone.png";
import loader from "@assets/dummy-images/product-thumbnails/loader.png";
import macbook from "@assets/dummy-images/product-thumbnails/macbook.png";
import mouse from "@assets/dummy-images/product-thumbnails/mouse.png";

export const generateArrowPanelData = (): IArrowPanelProps[] => {
  const data: IArrowPanelProps[] = [
    {
      image: iears,
      link: { url: "/", title: "Apple Airpods" }
    },
    {
      image: imac,
      link: { url: "/", title: "Apple iPhones" }
    },
    {
      image: iphone,
      link: { url: "/", title: "Apple MacBooks" }
    },
    {
      image: loader,
      link: { url: "/", title: "Apple iMacs" }
    },
    {
      image: macbook,
      link: { url: "/", title: "Categorie item met een lange naam" }
    },
    {
      image: mouse,
      link: { url: "/", title: "Categorie item met een nog langere naam" }
    },
    {
      link: { url: "/", title: "Categorie item zonder afbeelding" }
    },
    {
      icon: party,
      link: { url: "/", title: "Categorie met icoon" }
    }
  ];

  return data;
};
