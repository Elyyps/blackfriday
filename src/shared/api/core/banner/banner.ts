import { ImageType } from "@app/core/banner";

export class Banner {
  public buttonLink: string;
  public buttonTitle: string;
  public image?: ImageType;
  public imageLink?: string;
  public label: string;
  public logo: ImageType;
  public moreInfoLink: string;
  public showAlternativeBanner: boolean;
  public text: any;
  public title: string;
}
