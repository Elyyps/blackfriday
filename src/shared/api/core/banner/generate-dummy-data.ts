import { Banner } from "./banner";
import CoolBlue from "@assets/cool-blue.svg";

export const generateDummyBannerComponentData = (): Banner => ({
  title: "Black Friday bij Coolblue",
  buttonTitle: "Naar deals",
  buttonLink: "/deals-overview",
  image: {
    src: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/ad.png",
    alt: "Ad picture"
  },
  imageLink: "https://coolblue.nl",
  label: "22 t/m 26 nov.",
  logo: { src: CoolBlue, alt: "CoolBlue logo" },
  text: "Een selectie van bizarre aanbiedingen.",
  moreInfoLink: "https://coolblue.nl",
  showAlternativeBanner: false
});
