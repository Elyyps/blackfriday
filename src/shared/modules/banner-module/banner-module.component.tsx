import * as React from "react";
import { BannerComponent } from "@app/core/banner";
import { BannerModule } from "@app/api/modules/banner/banner.module";

export interface IBannerModuleComponentProps {
  bannerModule: BannerModule;
}

const BannerModuleComponent = (props: IBannerModuleComponentProps) => (
  <React.Fragment>
    <BannerComponent {...props.bannerModule.bannerComponentProps} />
  </React.Fragment>
);

export { BannerModuleComponent };
