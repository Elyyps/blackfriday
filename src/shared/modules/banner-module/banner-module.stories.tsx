import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { BannerModuleComponent } from "./banner-module.component";

// Dummy data
import { bannerProps } from "@app/api/core/banner";

storiesOf("Banner Module", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <BannerModuleComponent bannerProps={bannerProps} />);
