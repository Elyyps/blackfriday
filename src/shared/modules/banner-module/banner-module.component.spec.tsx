import React from "react";
import { shallow } from "enzyme";
import { BannerModuleComponent } from "./banner-module.component";

// Dummy data
import { generateDummyBannerModule } from "@app/api/modules/banner/generate-dummy-data";

describe("[BannerModule]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<BannerModuleComponent bannerModule={generateDummyBannerModule()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
