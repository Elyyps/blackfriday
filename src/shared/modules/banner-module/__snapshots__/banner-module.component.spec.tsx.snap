// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[BannerModule] should render component without crashing 1`] = `
<Fragment>
  <InjectIntl(component)
    buttonLink="/deals-overview"
    buttonTitle="Meer deals"
    image={
      Object {
        "alt": "alt text oasis",
        "src": "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/ad.png",
      }
    }
    imageLink="https://coolblue.nl"
    label="22 t/m 26 nov."
    logo={
      Object {
        "alt": "alt text CoolBlue",
        "src": "cool-blue.svg",
      }
    }
    moreInfoLink="https://coolblue.nl"
    showAlternativeBanner={false}
    text="Een selectie van bizarre aanbiedingen."
    title="Black Friday bij Coolblue"
  />
</Fragment>
`;
